examples/js/loaders/GLTFLoader.js

```diff
@@ -12,368 +16,1836 @@ THREE.GLTFLoader.prototype = {
+
+		_each( json.materials, function( material, materialId ) {
+
+			var materialType;
+			var materialValues = {};
+			var materialParams = {};
+
+			var khr_material;
+
+			if ( material.extensions && material.extensions.KHR_materials_common ) {
+
+				khr_material = material.extensions.KHR_materials_common;
+
+			} else if ( json.extensions && json.extensions.KHR_materials_common ) {
+
+				khr_material = json.extensions.KHR_materials_common;
+
+			}
+
+			if ( khr_material ) {
+
+				switch ( khr_material.technique )
+				{
+					case 'BLINN' :
+					case 'PHONG' :
+						materialType = THREE.MeshPhongMaterial;
+						break;
+
+					case 'LAMBERT' :
+						materialType = THREE.MeshLambertMaterial;
+						break;
+
+					case 'CONSTANT' :
+					default :
+						materialType = THREE.MeshBasicMaterial;
+						break;
+				}
+
+				_each( khr_material.values, function( value, prop ) {
+
+					materialValues[ prop ] = value;
+
+				});
+
+				if ( khr_material.doubleSided )
+				{
+
+					materialParams.side = THREE.DoubleSide;
+
+				}
+
+				if ( khr_material.transparent )
+				{
+
+					materialParams.transparent = true;
+
+				}
+
+			} else if ( material.technique === undefined ) {
+
+				materialType = THREE.MeshPhongMaterial;
+
+				_each( material.values, function( value, prop ) {
+
+					materialValues[ prop ] = value;
+
+				});
+
+			} else {
+
+				materialType = DeferredShaderMaterial;
+
+				var technique = json.techniques[ material.technique ];
+
+				materialParams.uniforms = {};
+
+				var program = json.programs[ technique.program ];
+
+				if ( program ) {
+
+					materialParams.fragmentShader = library.shaders[ program.fragmentShader ];
+
+					if ( ! materialParams.fragmentShader ) {
+
+						console.warn( "ERROR: Missing fragment shader definition:", program.fragmentShader );
+						materialType = THREE.MeshPhongMaterial;
+
+					}
+
+					var vertexShader = library.shaders[ program.vertexShader ];
+
+					if ( ! vertexShader ) {
+
+						console.warn( "ERROR: Missing vertex shader definition:", program.vertexShader );
+						materialType = THREE.MeshPhongMaterial;
+
+					}
+
+					// IMPORTANT: FIX VERTEX SHADER ATTRIBUTE DEFINITIONS
+					materialParams.vertexShader = replaceTHREEShaderAttributes( vertexShader, technique );
+
+					var uniforms = technique.uniforms;
+
+					_each( uniforms, function( pname, uniformId ) {
+
+						var shaderParam = technique.parameters[ pname ];
+
+						var ptype = shaderParam.type;
+
+						if ( WEBGL_TYPE[ ptype ] ) {
+
+							var pcount = shaderParam.count;
+							var value = material.values[ pname ];
+
+							var uvalue = new WEBGL_TYPE[ ptype ]();
+							var usemantic = shaderParam.semantic;
+							var unode = shaderParam.node;
+
+							switch ( ptype ) {
+
+								case WEBGL_CONSTANTS.FLOAT:
+
+									uvalue = shaderParam.value;
+
+									if ( pname == "transparency" ) {
+
+										uvalue = value;
+										materialParams.transparent = true;
+
+									}
```

Comment:
Thank you for fix. However, it looks like the code https://github.com/mrdoob/three.js/commit/e0e7dbf9fefa8ee6a01e580ca0801fd2c9bf0f40 has not been fixed.
If you did not add the "else" statement, uvalue is undefined.

For example, following code was extracted from duck.gltf.

```javascript
    "materials": {
        "blinn3-fx": {
            "technique": "technique0",
            "values": {
                "shininess": 38.4,
            }
        }
    },
    "techniques": {
        "technique0": {
            "parameters": {
                "shininess": {
                    "type": 5126
                },
```
When you load the above file, it will be in the following result.

```javascript
case WEBGL_CONSTANTS.FLOAT: // 5126

    uvalue = shaderParam.value; // shaderParam.value is undefined (= technique.parameters["shininess"].value)

    if ( pname == "transparency" ) {
        uvalue = value;
        materialParams.transparent = true;
    } else { // please add the "else" statement
        uvalue = value;  // value is 38.4
    }

    break;
```